File: src/api/community.js
import client from './client';

function normalizeFanWar(war) {
  return {
    id: war.id,
    title: war.title,
    sport: war.sport,
    status: war.status,
    teamA: war.teamA || war.team_a,
    teamB: war.teamB || war.team_b,
  };
}

function normalizeDebate(debate) {
  return {
    ...debate,
    optionA: debate.optionA || debate.option_a,
    optionB: debate.optionB || debate.option_b,
  };
}

export const communityApi = {
  async getFanDNA() {
    const { data } = await client.get('/community/fan-dna');
    return data;
  },

  async getDebates() {
    const { data } = await client.get('/community/debates');
    return (data.items || []).map(normalizeDebate);
  },

  async getFanWars() {
    const { data } = await client.get('/community/fan-wars');
    return (data.items || []).map(normalizeFanWar);
  },

  async voteFanWar(warId, side) {
    const { data } = await client.post('/community/fan-wars/vote', { war_id: warId, side });
    return data;
  },

  async getPredictions() {
    const { data } = await client.get('/community/predictions');
    return {
      events: data.events || [],
      leaderboard: data.leaderboard || [],
    };
  },

  async submitPrediction(eventId, answers) {
    const { data } = await client.post('/community/predictions', { event_id: eventId, answers });
    return data;
  },

  async getLeaderboard() {
    const { data } = await client.get('/community/leaderboard');
    return data.items || [];
  },
};

File: backend/app/api/v1/community.py
from typing import Any

from fastapi import APIRouter, Depends
from pydantic import BaseModel
from sqlalchemy.orm import Session

from app.core.deps import get_optional_user
from app.db.session import get_db
from app.models import User
from app.services.community_service import CommunityService

router = APIRouter(prefix="/community", tags=["Fan Community"])


class FanWarVoteRequest(BaseModel):
    war_id: str
    side: str


class PredictionRequest(BaseModel):
    event_id: str
    answers: dict[str, Any]


@router.get("/fan-dna")
def get_fan_dna(user: User | None = Depends(get_optional_user)):
    return CommunityService.get_fan_dna(user.id if user else None)


@router.get("/debates")
def get_debates(db: Session = Depends(get_db)):
    return {"items": CommunityService.get_debates(db)}


@router.get("/fan-wars")
def get_fan_wars(db: Session = Depends(get_db)):
    return {"items": CommunityService.get_fan_wars(db)}


@router.post("/fan-wars/vote")
def vote_fan_war(
    data: FanWarVoteRequest,
    db: Session = Depends(get_db),
    user: User | None = Depends(get_optional_user),
):
    return CommunityService.vote_fan_war(db, data.war_id, data.side, user.id if user else None)


@router.get("/predictions")
def get_predictions(db: Session = Depends(get_db)):
    return CommunityService.get_predictions(db)


@router.post("/predictions")
def submit_prediction(
    data: PredictionRequest,
    db: Session = Depends(get_db),
    user: User | None = Depends(get_optional_user),
):
    return CommunityService.submit_prediction(db, user.id if user else None, data.event_id, data.answers)


@router.get("/leaderboard")
def get_leaderboard(db: Session = Depends(get_db), user: User | None = Depends(get_optional_user)):
    return {"items": CommunityService.get_leaderboard(db, user.id if user else None)}
